### PR TITLE
Financial Connections: for V3, got a basic first-pass design for consent pane

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR modifies the design of Financial Connections consent pane.

Details to understand before reviewing PR:
- This is one PR, out of expected ~50+, to change the design of Financial Connections from a "V2" design to a "V3" design. [Here is a link to the designs](https://www.figma.com/file/wXQ8NJApqSJiLmxxANDRTs/%E2%9C%A8-Connections-3.0?type=design&node-id=2524-217093&mode=design&t=kXaBiAlgVsC7YcH1-0).
- The code here is not the finished version. It's just an iteration towards the final version. This code will be re-visited later. The goal is to go screen-by-screen and make easy changes to learn about all the difficulties in each screen. Later, all the learnings will be combined to do a "final" polish of all screens.
- The above also applies to design. The design here is not the finished version.
- This PR is merged into a feature branch `fc-v3`. It is not merged into `master`.
- There might be TODO's scattered throughout the code. That's expected and they will be fixed later. 

https://jira.corp.stripe.com/browse/BANKCON-8504

## Testing

The rendering of the consent pane still works and is modified:

https://github.com/stripe/stripe-ios/assets/105514761/10cf982a-bddf-45b7-868f-70f87fcb6612

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Consent/ConsentLogoView.swift [hunk 1]
```diff
@@ -65,12 +64,266 @@ private func CreateCircularLogoView(urlString: String) -> UIView {
...
+    multipleDotView.frame = CGRect(
+        x: -multipleDotView.bounds.width + ellipsisViewWidth,
+        y: 0,
+        width: multipleDotView.bounds.width,
+        height: multipleDotView.bounds.height
+    )
+
+    backgroundView.mask = multipleDotView
+
+    // TODO(kgaidis): add code to repeat animation even when UIViewController is presented
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/Consent/ConsentLogoView.swift:94]: Note that the TODO's are expected, please see PR description where a lot of this code is WIP as part of an integration branch:
- https://github.com/stripe/stripe-ios/pull/3093#issue-2014979892

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/FinancialConnectionsFont.swift [hunk 2]
```diff
@@ -26,6 +26,8 @@ struct FinancialConnectionsFont {
         case medium
         /// 24 size / 32 line height / 700 weight
         case large
+        /// 28 size / 36 line height / 600 weight
```
Comment (ccen-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/FinancialConnectionsFont.swift:29]: is 600 weight expected for extra large?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift [hunk 3]
```diff
@@ -71,16 +79,20 @@ extension UIColor {
...
+        return UIColor(red: 53 / 255.0, green: 58 / 255.0, blue: 68 / 255.0, alpha: 1)  // #353a44
     }
 
     static var brand100: UIColor {
         return UIColor(red: 242 / 255.0, green: 235 / 255.0, blue: 255 / 255.0, alpha: 1)  // #f2ebff
     }
 
     private static var brand500: UIColor {
-        return UIColor(red: 99 / 255.0, green: 91 / 255.0, blue: 255 / 255.0, alpha: 1)  // #635bff
+        return UIColor(red: 103 / 255.0, green: 93 / 255.0, blue: 255 / 255.0, alpha: 1)  // #675DFF
```
Comment (ccen-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift:95]: others seems lower casesd, should this be lower case too?

**Review by ccen-stripe — APPROVED** [review]
LGTM! do we have screenshot tests to cover these UI changes?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift (hunk 3 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift:95]: good attention to detail 💯 I'll change to lowercase

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/FinancialConnectionsFont.swift (hunk 2 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/FinancialConnectionsFont.swift:29]: thank you for noticing, I double-checked and changed everything to 600

**Pull request merged**